### PR TITLE
feat(datafile): Support for DatafileURLTemplate.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Added support to pass through DatafileURLTemplate from the Agent config to the Go SDK project config manager.

## Test Plan
- FSC should pass.
- Updated unit tests accordingly.

## Issues
- https://optimizely.atlassian.net/browse/OASIS-6329

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -91,3 +91,5 @@ client:
     queueSize: 1000
     ## the maximum time between events being dispatched
     flushInterval: 30s
+    ## the endpoint for retrieving the datafile from the CDN
+    datafileURLTemplate: "https://cdn.optimizely.com/datafiles/%s.json"
```
Comment (mikeng13) [inline config.yaml:95]: This is defined in the Go SDK publicly here: https://github.com/optimizely/go-sdk/blob/master/pkg/config/polling_manager.go#L4 please reference that instead

Comment (msohailhussain) [reply config.yaml:95]: eol

File: config.yaml [hunk 2]
```diff
@@ -91,3 +91,5 @@ client:
     queueSize: 1000
     ## the maximum time between events being dispatched
     flushInterval: 30s
+    ## the endpoint for retrieving the datafile from the CDN
```
Comment (mikecdavis) [inline config.yaml:94]: Let's use this to describe the formatting.
> Template URL for SDK datafile location. The template should specify a "%s" token for SDK key substitution.

File: cmd/optimizely/main_test.go [hunk 3]
```diff
@@ -215,6 +217,7 @@ func TestViperEnv(t *testing.T) {
 	_ = os.Setenv("OPTIMIZELY_CLIENT_BATCHSIZE", "1")
 	_ = os.Setenv("OPTIMIZELY_CLIENT_QUEUESIZE", "10")
 	_ = os.Setenv("OPTIMIZELY_CLIENT_FLUSHINTERVAL", "1m")
+	_ = os.Setenv("OPTIMIZELY_CLIENT_DATAFILEURLTEMPLATE", "https://localhost/v1/%s.json")
```
Comment (msohailhussain) [inline cmd/optimizely/main_test.go:220]: @mikecdavis 
Shouldn't it be `DATAFILE_URL_TEMPLATE`

Comment (mikecdavis) [reply cmd/optimizely/main_test.go:220]: No this needs to be `DATAFILEURLTEMPLATE` based on how viper maps config values to ENV variables.

**Pull request merged**